### ISSUE TITLE
Errors in calculating the geometry of windows

### ISSUE THREAD

**Issue by arsonoff** [bbidulock/icewm#75]
Run the three programs (xclock, mplayer, gimp)
Make them the same size (about 640x480).
Compare posts xwininfo and icewm (tooltip appearing when moving windows):

```
Program    xwininfo    icewm
xclock     640x480     640x480
smplayer   638x481     238x234
gimp       641x479     414x408
```

The big problem with VirtualBox:
Specifies a display resolution of 640x480 inside a virtual machine.
IceWM window size changes, but not right (the window is less than required).
xwininfo:

```
  …
  Corners:  +750+363  -216+363  -216-342  +750-342
  -geometry 634x495+750+363
```

icewm (tooltip): 344x397
xprop:

```
…
WM_NORMAL_HINTS(WM_SIZE_HINTS):
        user specified location: 842, 305
        program specified location: 842, 305
        user specified size: 640 by 521
        program specified size: 640 by 521
        program specified minimum size: 290 by 98
        window gravity: Static
…
```

If set manually (mouse) window dimensions 640x521 , everything fits.
If set the option in VB «Auto-resize Guest Display», the window is compressed in the cycle (up to size zero).

**Comment by arsonoff** [comment]
Icewm tooltip shows the size of a inner window (not necessarily the main), sizes vary depending on the position of the widgets within the window.

**Comment by bbidulock** [comment]
I notice that your windows are requesting StaticGravity.  According to ICCCM 2.0 the window manager is correct to override this with NorthwestGravity if it feels like it.  Some toolkits might not be observing that.  I strapped out the override to NorthwestGravity seeing as icewm appears capable of handling StaticGravity.  The override might have been messing up some calculations (icewm thinks it is StaticGravity when I overrode to NorthwestGravity).  So the commit stands a chance of fixing the behavior.  Please test.

**Comment by arsonoff** [comment]
> Please test.

unchanged (window less than required)

Example with a resolution of 640x480 in VirtualBox is not correct. In other resolutions xprop still returns "user/ program specified size: 640 by 521" (???)

In 1.3.8 size changes correctly.

**Comment by bbidulock** [comment]
Ok, thank you.  I check into it a little further.

**Comment by bbidulock** [comment]
Fixed by commit  2b0df4d

**Comment by arsonoff** [comment]
> Please test.

Sorry, I thought about the problem with the VB.
Tooltips now displays the correct size. OK.
